### PR TITLE
Todo-17-BE: As a user, I want to select all todos instantly and get warning when trying to delete anything

### PR DESCRIPTION
**Pull request by itc02**

### CODE REVIEW

File: app/controllers/todos_controller.rb [hunk 1]
```diff
@@ -1,5 +1,13 @@
 class TodosController < ApplicationController
   def index
+    if params[:all_todos_id]
```
Comment (arpinehp) [inline app/controllers/todos_controller.rb]: `all_todos_ids`
use switch cases

File: app/controllers/todos_controller.rb [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
-      :page => params[:page],
-      :sorting_criteria => params[:sorting_criteria],
-      :order => params[:order],
-      :search_string => params[:search_string],
-      :filter_criteria => params[:filter_criteria]
-    )
+    case
+      when params[:all_todos_ids] then render :json => GetAllTodosIdService.run!()
+      when params[:id] then render :json => GetTodoByIdService.run!(:id => params[:id])
+      else
```
Comment (arpinehp) [inline app/controllers/todos_controller.rb:8]: default?

File: app/controllers/todos_controller.rb [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
-    todos = GetTodosService.run(
-      :per => params[:per],
-      :page => params[:page],
-      :sorting_criteria => params[:sorting_criteria],
-      :order => params[:order],
-      :search_string => params[:search_string],
-      :filter_criteria => params[:filter_criteria]
-    )
+    case
+      when params[:all_todos_ids] then render :json => GetAllTodosIdService.run!()
```
Comment (arpinehp) [inline app/controllers/todos_controller.rb]: keep the coding style.

File: app/services/get_todo_by_id_service.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class GetTodoByIdService < ActiveInteraction::Base
+  integer :id
+
+  def execute
+    TodoList.find(id)
+  end
+end
```
Comment (arpinehp) [inline app/services/get_todo_by_id_service.rb:7]: add a newline at the end of file

Comment (itc02) [reply app/services/get_todo_by_id_service.rb:7]: I created separate ticket for it.

File: app/controllers/todos_controller.rb (hunk 2 above)
Comment (itc02) [reply app/controllers/todos_controller.rb:8]: Isn't `else` the equivalent for `default` in Ruby?

File: app/controllers/todos_controller.rb (hunk 3 above)
Comment (itc02) [reply app/controllers/todos_controller.rb]: Should I write like this:
```
when params[:all_todos_ids]
    render :json => GetAllTodosIdService.run!()
```
?

Comment (arpinehp) [reply app/controllers/todos_controller.rb]: Correct!

File: app/controllers/todos_controller.rb (hunk 2 above)
Comment (arpinehp) [reply app/controllers/todos_controller.rb:8]: yes, sorry :facepalm:  my bad

**Review by arpinehp — APPROVED** [review]

**Pull request merged**